library/entity/points_of_interest/src/main/java/org/quiltmc/qsl/points_of_interest/impl/PointOfInterestTypeExtensions.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.points_of_interest.impl;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.world.poi.PointOfInterest;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import java.util.Collection;
+
+import org.quiltmc.qsl.base.api.util.InjectedInterface;
+
+@ApiStatus.Internal
+@InjectedInterface(PointOfInterestType.class)
```

Comment:
`@InjectedInterface`s should be used for _public_ extensions.
This should be a duck interface (implemented by `PointOfInterestMixin` directly).

Reply:
Accordingly, all of the methods here should be prefixed with `quilt$`.